readme.txt
/**
* $Id: readme.txt,v 1.7 2006/10/23 19:37:16 malanciault Exp $
* Module: FreeSection
* Author: The SmartFactory <www.smartfactory.ca>; adapted by Howtodude.NET
* Licence: GNU
*/

What is FreeSection ?
======================
FreeSection is a derivate of SmartSection. 
- it was cloned
- It has a hidden text link removed as described in http://www.howtodude.net/howto/view.article.php/167
- it has removed the page title as described in http://www.howtodude.net/howto/view.article.php/197


What is SmartSection ?
======================

SmartSection is an articles management module for your XOOPS 2.x site. It is intended to provide the best functionnalities of the News and WF-Section modules. One would have the opportunity to use the module as a news management system, as a section management system, or maybe both ! It is intended to be complete, flexible and yet easy to use.

How to install SmartSection
============================

SmartSection is installed as a regular XOOPS module, which means you should copy the complete /smartsection folder into the /modules directory of your website. Then log in to your site as administrator, go to System Admin > Modules, look for the SmartFAQ icon in the list of uninstalled modules and click in the install icon. Follow the directions in the screen and you'll be ready to go.


Feedback
========

This module uses the Developpers XOOPS Forge for its developpement.

=> Please post any bug you may found here : http://dev.xoops.org/modules/xfmod/tracker/?func=add&group_id=1102&atid=546

=> Please  post any feature request here : http://dev.xoops.org/modules/xfmod/tracker/?func=add&group_id=1102&atid=549


.:: The SmartFactory ::.
.:: Open Source : smartfactory.ca ::.
.:: Professionnal : inboxinternational.com ::.


changelog.txt
<b><u>=> FreeSection Version 2.13 Final (unreleased)</u></b>
- cloned SmartSection according to this howto: http://www.howtodude.net/howto/view.article.php/167
- removed module name from title according to this howto: http://www.howtodude.net/howto/view.article.php/197
- removed hidden text links according to this howto: http://www.howtodude.net/howto/view.article.php/167
- updated xoopsinfo.php
- updated changelog.txt

<b><u>=> Version 2.13 Final (unreleased)</u></b>

- Added : the article body now supports the [flash-fileid] tag where fileid is the id of a flash file linked to an article. This tag will be automatically replaced with the embeded Flash.
- Fixed : a user who is notauthor of an article was able to link a file to it under som circumstances
- Fixed : Undefined variable: totalItemsOnPage in file modules/smartsection/index.php line 177
- Fixed : Smarty error: unable to read resource: "db:tag_bar.html"
- Fixed : Undefined variable: max_imgsize in file modules/smartsection/include/fileform.inc.php line 42
- Fixed : Object of class SmartsectionCategory could not be converted to int in file modules/smartsection/class/item.php line 179
- Fixed : Object of class SmartsectionCategory could not be converted to int in file modules/smartsection/class/item.php line 181
- Fixed : Object of class SmartsectionCategory could not be converted to int in file language/german/local.php line 11
- Fixed : Object of class SmartsectionCategory could not be converted to int in file language/german/local.php line 11
- Fixed : Object of class SmartsectionCategory could not be converted to int in file modules/smartsection/include/functions.php line 161
- Fixed : ' char was causing a problem in the title of the article as the TITLE tag of the page

<b><u>=> Version 2.13 RC 4(forum release)</u></b>

- Fixed : security issue fixed thanks to Rroselicoe;

<b><u>=> Version 2.13 RC 4(forum release)</u></b>

- Fixed : missing item_tag field on clean install
- Fixed : undefined constant _MI_SSECTION_LATESTFILES
- Fixed : Undefined offset: 3 in file modules/smartsection/blocks/items_spot.php line 20

<b><u>=> Version 2.13 RC 3(forum release)</u></b>

- Added : option to set the title length in recent items and new items block
- Fixed : Tag handler does not exists
- Fixed : some problems occured when the Tag module was not installed

<b><u>=> Version 2.13 RC 2(forum release)</u></b>

- Added : it is now possible to attach a fil to an article directly from the creation/edition form

<b><u>=> Version 2.13 RC 1(2007-01-29)</u></b>

- Fixed : In userinfo page, when no keywords were looked for, the article link still had the &keywords string which was screwing the layout on the article page
- Removed : removing class/metagen.php as it is now handle by smartmetagen.php
- Added : we are now wupporting phpPP's Tags module
- Added : When a SWF file is linked to an article, the SWF will be dipsplayed embed in the article
- Added : Category images are now displayed within the Summary view and Full view display type
- Added : When a category does not hold any articles, the article count is not displayed, instead of being displayed as 0

<b><u>=> Version 2.12 RC2 (unreleased)</u></b>

- Fixed : SmartMetaGen class was replaced by SmartsectionMetaGen
- Fixed : Spotlight block was sorting ASC by date instead of DESC by date
- Fixed : french language constant in block (regarding the spotlight block) was fixed
- Fixed : Undefined offset: 63 in file /modules/smartsection/class/smartmetagen.php line 265
- Fixed : Fatal error: Call to undefined function smartsection_getModuleConfig()
- Fixed : Call to a member function setDescription() on a non-object in smartsection\admin\item.php on line 227 when a (some) non existing WYSIWYG editor was selected
- Fixed : Only variables should be assigned by reference in file /modules/smartsection/class/permission.php line 36
- Fixed : Only variables should be assigned by reference in file /modules/smartsection/admin/item.php line 168
- Fixed : Only variables should be assigned by reference in file /modules/smartsection/admin/item.php line 272
- Fixed : Only variables should be assigned by reference in file /modules/smartsection/admin/item.php line 351
- Fixed : Only variables should be assigned by reference in file /modules/smartsection/xoops_version.php line 622
- Fixed : Only variables should be assigned by reference in file /modules/smartsection/class/item.php line 603
- Fixed : It was only possible to configure the Partial View for 1 group
- Added : A new config option was added to enable category breadcrumb in the search results

<b><u>=> Version 2.12 RC1 (2006-10-19)</u></b>

- Added : New block to list the latest uploaded files
- Added : more inmprovment to the SmartMetaGen thanks to PsyLove

<b><u>=> Version 2.11 RC1 (2006-09-29)</u></b>

- Added : Support for Importing News 1.44
- Added : Sanitizig accentuated characters in the short_url thanks to psylove
- Added : module config option to personnalized the name of the module when using URL Rewriting
- Fixed : Send to a friend button was having a blank body when the SiteName had an apostrophe
- Fixed : Wrong file inclusion on admin/pw_upload_file an class/file.php, thanks to dead
- Fixed : problem with articles having categoryid=0. This is fixed when updating the module (articles that have caegoryid=0 will be changed for an available categoryid)
- Fixed : Wrong URL for attached files when using URL_REWRITE
- Fixed : DisplayLastItems config option was no more present in the preferences
- Fixed : undefined variable - QUERY_STRING in include/functions.php thanks to Peekay
- Fixed : Changing INSERT queries to fix problems related to MySQL 5.x
- Changed : TinyEditor has changed its code so SmartSection needed to update its support for this editor (thanks to alain01)
- Changed : The print and pdf link of an article are no more opening in a new window

item.php
<?php

/**
* $Id: item.php,v 1.51 2007/02/09 16:02:56 malanciault Exp $
* Module: FreeSection
* Author: The SmartFactory <www.smartfactory.ca>
* Licence: GNU
*/

include_once("header.php");

global $freesection_item_handler, $freesection_category_handler, $xoopsUser, $xoopsConfig, $xoopsModuleConfig, $xoopsModule;

$itemid = isset($_GET['itemid']) ? intval($_GET['itemid']) : 0;
$item_page_id = isset($_GET['page']) ? intval($_GET['page']) : -1;

if ($itemid == 0) {
	redirect_header("javascript:history.go(-1)", 1, _MD_FSECTION_NOITEMSELECTED);
	exit();
}

// Creating the item object for the selected item
$itemObj = $freesection_item_handler->get($itemid);

// if the selected item was not found, exit
if (!$itemObj) {
	redirect_header("javascript:history.go(-1)", 1, _MD_FSECTION_NOITEMSELECTED);
	exit();
}

$xoopsOption['template_main'] = 'freesection_item.html';
include_once(XOOPS_ROOT_PATH . "/header.php");
include_once("footer.php");

// Creating the category object that holds the selected item
$categoryObj =& $freesection_category_handler->get($itemObj->categoryid());

// Check user permissions to access that category of the selected item
if (!(freesection_itemAccessGranted($itemObj->itemid(), $itemObj->categoryid()))) {
	redirect_header("javascript:history.go(-1)", 1, _NOPERM);
	exit;
}

// Update the read counter of the selected item
if (!$xoopsUser || ($xoopsUser->isAdmin($xoopsModule->mid()) && $xoopsModuleConfig['adminhits'] == 1) || ($xoopsUser && !$xoopsUser->isAdmin($xoopsModule->mid()))) {
	$itemObj->updateCounter();
}

// creating the Item objects that belong to the selected category
switch ($xoopsModuleConfig['orderby']) {
	case 'title' :
    	$sort = 'title';
    	$order = 'ASC';
	break;

	case 'date' :
    	$sort = 'datesub';
    	$order = 'DESC';
	break;

	default :
    	$sort = 'weight';
    	$order = 'ASC';
	break;
}

$itemsObj = $freesection_item_handler->getAllPublished(0, 0, $categoryObj->categoryid(), $sort, $order, '', true, true);

// Retreiving the next and previous object
$array_keys = array();
foreach ($itemsObj as $key=> $item){
	$array_keys[$key]= $item->itemid();
}

$current_item = array_search($itemid, $array_keys);
$items_count = count($array_keys);
$previous_item = $current_item - 1;
$next_item = $current_item + 1;


if ($previous_item >= 0) {
	$previous_item_link = $itemsObj[$previous_item]->getItemLink();
	$previous_item_url = $itemsObj[$previous_item]->getItemUrl();
} else {
	$previous_item_link = '';
	$previous_item_url = '';
}
if ($next_item < $items_count) {
	$next_item_link = $itemsObj[$next_item]->getItemLink();
	$next_item_url = $itemsObj[$next_item]->getItemUrl();
} else {
	$next_item_link = '';
	$next_item_url = '';
}

// Populating the smarty variables with informations related to the selected item
$item = $itemObj->toArray($item_page_id);
$item['who_when'] = $itemObj->getWhoAndWhen();

if ($itemObj->pagescount() > 0) {
	include_once XOOPS_ROOT_PATH.'/class/pagenav.php';
	if ($item_page_id == -1) ($item_page_id = 0);
    $pagenav = new XoopsPageNav($itemObj->pagescount(), 1, $item_page_id, 'page', 'itemid=' . $itemObj->itemid());
    $xoopsTpl->assign('pagenav', $pagenav->renderNav());
}


$items = array();

foreach($itemsObj as $theitemObj)
{
	$theitem['titlelink'] = $theitemObj->getItemLink();
	$theitem['datesub'] = $theitemObj->datesub();
	$theitem['counter'] = $theitemObj->counter();
	if ($theitemObj->itemid() == $itemObj->itemid()) {
		$theitem['titlelink'] = $theitemObj->title();
	}
	$items[] = $theitem;
	unset($theitem);
}
$xoopsTpl->assign('items', $items);

// Creating the files object associated with this item
$filesObj = $itemObj->getFiles();

$files = array();
$embeded_files = array();

foreach($filesObj as $fileObj)
{
	if ($fileObj->mimetype() == 'application/x-shockwave-flash') {
		$file['content'] = $fileObj->displayFlash();

		if (strpos($item['maintext'], '[flash-' . $fileObj->getVar('fileid') . ']')) {
			$item['maintext'] = str_replace('[flash-' . $fileObj->getVar('fileid') . ']', $file['content'], $item['maintext']);
		} else {
			$embeded_files[] = $file;
		}
		unset($file);
	} else {
		$file['fileid'] = $fileObj->fileid();
		$file['name'] = $fileObj->name();
		$file['description'] = $fileObj->description();
		$file['name'] = $fileObj->name();
		$file['type'] = $fileObj->mimetype();
		$file['datesub'] = $fileObj->datesub();
		$file['hits'] = $fileObj->counter();
		$files[] = $file;
		unset($file);
	}

}
$item['files'] = $files;
$item['embeded_files'] = $embeded_files;

// Language constants
$xoopsTpl->assign('item', $item);
$xoopsTpl->assign('mail_link', 'mailto:?subject=' . sprintf(_MD_FSECTION_INTITEM, $xoopsConfig['sitename']) . '&amp;body=' . sprintf(_MD_FSECTION_INTITEMFOUND, $xoopsConfig['sitename']) . ': ' . $itemObj->getItemUrl());
$xoopsTpl->assign('lang_printerpage', _MD_FSECTION_PRINTERFRIENDLY);
$xoopsTpl->assign('lang_sendstory', _MD_FSECTION_SENDSTORY);
$xoopsTpl->assign('itemid', $itemObj->itemid());
$xoopsTpl->assign('sectionname', $freesection_moduleName);
$xoopsTpl->assign('modulename', $xoopsModule->dirname());
$xoopsTpl->assign('lang_home', _MD_FSECTION_HOME);
$xoopsTpl->assign('lang_item', _MD_FSECTION_OTHER_ITEMS);
$xoopsTpl->assign('lang_postedby', _MD_FSECTION_POSTEDBY);
$xoopsTpl->assign('lang_on', _MD_FSECTION_ON);
$xoopsTpl->assign('lang_datesub', _MD_FSECTION_DATESUB);
$xoopsTpl->assign('lang_hitsdetail', _MD_FSECTION_HITSDETAIL);
$xoopsTpl->assign('lang_reads', _MD_FSECTION_READS);
$xoopsTpl->assign('lang_comments', _MD_FSECTION_COMMENTS);
$xoopsTpl->assign('lang_files_linked', _MD_FSECTION_FILES_LINKED);
$xoopsTpl->assign('lang_file_name', _MD_FSECTION_FILENAME);
$xoopsTpl->assign('lang_file_type', _MD_FSECTION_FILE_TYPE);
$xoopsTpl->assign('lang_hits', _MD_FSECTION_HITS);
$xoopsTpl->assign('lang_download_file', _MD_FSECTION_DOWNLOAD_FILE);
$xoopsTpl->assign('lang_page', _MD_FSECTION_PAGE);
$xoopsTpl->assign('lang_previous_item', _MD_FSECTION_PREVIOUS_ITEM);
$xoopsTpl->assign('lang_next_item', _MD_FSECTION_NEXT_ITEM);

$xoopsTpl->assign('module_home', freesection_module_home($xoopsModuleConfig['linkedPath']));
$xoopsTpl->assign('categoryPath', $item['categoryPath'] . " > " . $item['title']);
$xoopsTpl->assign('commentatarticlelevel', $xoopsModuleConfig['commentatarticlelevel']);
$xoopsTpl->assign('com_rule', freesection_getConfig('com_rule'));
$xoopsTpl->assign('lang_items_links', _MD_FSECTION_ITEMS_LINKS);
$xoopsTpl->assign('previous_item_link', $previous_item_link);
$xoopsTpl->assign('next_item_link', $next_item_link);
$xoopsTpl->assign('previous_item_url', $previous_item_url);
$xoopsTpl->assign('next_item_url', $next_item_url);
$xoopsTpl->assign('other_items', $xoopsModuleConfig['other_items_type']);

$itemFooter = freesection_getConfig('itemfooter');
$itemFooter = $myts->displayTarea($itemFooter);
$xoopsTpl->assign('itemfooter', $itemFooter);

// tags support
if (freesection_tag_module_included()) {
	include_once XOOPS_ROOT_PATH."/modules/tag/include/tagbar.php";
	$xoopsTpl->assign('tagbar', tagBar($itemid, $catid = 0));
}
/**
 * Generating meta information for this page
 */
$freesection_metagen = new FreesectionMetagen($itemObj->getVar('title'), $itemObj->getVar('meta_keywords','n'), $itemObj->getVar('meta_description', 'n'), $itemObj->getCategoryPath());
$freesection_metagen->createMetaTags();

// Include the comments if the selected ITEM supports comments
if (($itemObj->cancomment() == 1) || (!$xoopsModuleConfig['commentatarticlelevel'] && freesection_getConfig('com_rule') <> 0)) {
	include_once XOOPS_ROOT_PATH . "/include/comment_view.php";
// Problem with url_rewrite and posting comments :
$xoopsTpl->assign(array('editcomment_link' => FREESECTION_URL . 'comment_edit.php?com_itemid='.$com_itemid.'&amp;com_order='.$com_order.'&amp;com_mode='.$com_mode.''.$link_extra, 'deletecomment_link' => FREESECTION_URL . 'comment_delete.php?com_itemid='.$com_itemid.'&amp;com_order='.$com_order.'&amp;com_mode='.$com_mode.''.$link_extra, 'replycomment_link' => FREESECTION_URL . 'comment_reply.php?com_itemid='.$com_itemid.'&amp;com_order='.$com_order.'&amp;com_mode='.$com_mode.''.$link_extra));
$xoopsTpl->_tpl_vars['commentsnav'] = str_replace("self.location.href='", "self.location.href='" . FREESECTION_URL, $xoopsTpl->_tpl_vars['commentsnav']);
}

//code to include smartie
/*if (file_exists(XOOPS_ROOT_PATH . '/modules/smarttie/smarttie_links.php')) {
	include_once XOOPS_ROOT_PATH . '/modules/smarttie/smarttie_links.php';
		$xoopsTpl->assign('smarttie',1);
}*/
//end code for smarttie
include_once XOOPS_ROOT_PATH . '/footer.php';

?>